### PR TITLE
Update transformer version to 0.5.0

### LINKED ISSUE #20472: JakartaEE: Transformer: Bring in transformer 0.4.0
See:

https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/

Notes:

(1) Need to make sure it still works.
(2) Very probably will need to adjust to API changes.

Note the following, which build on the 0.3 update:

* https://github.com/eclipse/transformer/issues/247
  - "Add Bnd Analyzer plugin for transforming during a Bnd operation"

This should provide the function required for:
* https://github.ibm.com/websphere/WS-CD-Open/issues/25634
  - "Change to use a bnd plugin to do transformation instead of gradle plugin"

Note: By generating by a transformed bnd.bnd, the POM metadata should be automatically updated.  The following should no longer be needed:

* https://github.com/OpenLiberty/open-liberty/issues/18538
  - "Support for bundles (pom.xml)"

### PR DESCRIPTION
**Pull request by tbitonti**
Closes #20472

Update version to 0.5.0-20220411
Add dependency on org.eclipse.transformer.bnd.analyzer

Add dependency on org.eclipse.transformer.jakarta.
Update transformer versions to 0.4.0.20220406
Update commons-cli version from 1.4 to 1.5.0.
Update main class to "org.eclipse.transformer.cli.JakartaTransformerCLI".

This update is to build tooling.  No shippable code was updated.

### CODE REVIEW

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
#libby

**Comment by tbitonti** [comment]
Initial build is badly broken.  Trying again ...
#build

**Comment by tbitonti** [comment]
One more try ...
#build

File: dev/wlp-gradle/subprojects/fat.gradle [hunk 1]
```diff
@@ -48,22 +48,22 @@ dependencies {
...
-		'org.apache.derby:derbyclient:10.11.1.1',
-		'org.apache.derby:derby:10.11.1.1',
-		'com.oracle.database.jdbc.debug:ojdbc8_g:21.1.0.0'
+    'org.postgresql:postgresql:42.2.5',
+    'com.microsoft.sqlserver:mssql-jdbc:9.2.1.jre8',
+    'org.apache.derby:derbyclient:10.11.1.1',
+    'org.apache.derby:derby:10.11.1.1',
+    'com.oracle.database.jdbc.debug:ojdbc8_g:21.1.0.0'
 
   jakartaTransformer 'biz.aQute.bnd:biz.aQute.bnd.transform:5.3.0',
```
Comment (cbridgha) [inline dev/wlp-gradle/subprojects/fat.gradle:57]: why didn't we update the existing wrapper project to use the latest version? - and then keep the "project" dependency

**Comment by tbitonti** [comment]
Turns out changing the kernel classes is a problem.  Those are for improved error reporting, so I'm removing those from the update.  That is:
```
  dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/provisioning/ContentBasedLocalBundleRepository.java
  dev/wlp-featureTasks/src/com/ibm/ws/wlp/feature/tasks/FeatureBnd.java
```

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
Duplicate of PR #22193.

**Pull request closed without merging**